app/assets/javascripts/reducers/users.js

```diff
@@ -23,12 +23,45 @@ const SORT_DESCENDING = {
 
 export default function users(state = initialState, action) {
   switch (action.type) {
-    case RECEIVE_USERS: return {
+    case RECEIVE_USERS: {
+      // Get the sorting key if available from Redux store or else use last_name
+      let sort_key = state.sort.key || 'last_name';
+      /* Use the previousKey to determine the directions of sorting.
+         Note: Direction of the sorting only matter after the user/instructor sort the students/editors
+         list using the dropdown menu or one of the student/editors header and then navigates between
+         tabs and then a data refresh occurs(usually after one minute).
+      */
+      let previousKey = null;
+      // Initialize a variable to hold the user list.
+      let user_list = action.data.course.users;
+
+      // Check if any user in the user_list has 'real_name'
+      if (user_list.some(user => user.real_name)) {
+        // If any users have 'real_name', transform the 'real_name' into separate
+        // 'first_name' and 'last_name' properties and update the user list
+        user_list = transformUsers(user_list);
```

Comment:
why is this if check even required? If `real_name` doesn't exist, `transformUsers` shouldn't change anything.